nimble/host/src/ble_hs_hci_evt.c

```diff
@@ -323,16 +323,22 @@ ble_hs_hci_evt_le_conn_complete(uint8_t subevent, uint8_t *data, int len)
         /* enhanced connection event has the same information with these
          * extra fields stuffed into the middle */
         if (subevent == BLE_HCI_LE_SUBEV_ENH_CONN_COMPLETE) {
+#if !MYNEWT_VAL(BLE_HOST_BASED_PRIVACY)
             memcpy(evt.local_rpa, data + 12, BLE_DEV_ADDR_LEN);
             memcpy(evt.peer_rpa, data + 18, BLE_DEV_ADDR_LEN);
+#endif
             extended_offset = 12;
 #if MYNEWT_VAL(BLE_HOST_BASED_PRIVACY)
             /* RPA needs to be resolved here, as controller is not aware of the
              * address is RPA in Host based RPA  */
             if (ble_host_rpa_enabled()) {
                 uint8_t *local_id_rpa = ble_hs_get_rpa_local();
                 memcpy(evt.local_rpa, local_id_rpa, 6);
-
+            } else {
+                memcpy(evt.local_rpa, data + 12, BLE_DEV_ADDR_LEN);
+            }
+            
+            if (ble_hs_is_rpa(evt.peer_addr, evt.peer_addr_type)) {
```

Comment:
@prasad-alatkar You are correct, however if the peer address is not RPA we will still add it to the device record list after the call to `ble_rpa_replace_peer_params_with_rl` since `rl == NULL` even if not RPA. The check here would avoid that, which is what I though you wanted to avoid in the first PR? Let me know and I will remove if you want.

Reply:
@prasad-alatkar I'm ready to push the changes, before I do could you advise on this please?